### PR TITLE
Add modifyOutput function to TestFormula.

### PR DESCRIPTION
**Pull request by Laimiux**
## What
To simplify modifying a single property of the output, I'm adding `modifyOutput` function to `TestFormula`

```kotlin
val formula = TestFormula(initialOutput = Value(a, b))
formula.modifyOutput { copy(a = newA) }
```

### CODE REVIEW

File: formula-test/src/test/java/com/instacart/formula/test/SimpleFormula.kt [hunk 1]
```diff
@@ -0,0 +1,23 @@
+package com.instacart.formula.test
+
+import com.instacart.formula.IFormula
+import com.instacart.formula.test.SimpleFormula.Input
+import com.instacart.formula.test.SimpleFormula.Output
+
+interface SimpleFormula : IFormula<Input, Output> {
```
Comment (Laimiux) [inline formula-test/src/test/java/com/instacart/formula/test/SimpleFormula.kt:7]: Rewriting the tests using a much simpler formula.

File: formula-test/src/test/java/com/instacart/formula/test/TestFormulaRobot.kt [hunk 2]
```diff
@@ -1,72 +0,0 @@
...
-
-    fun withTestFormula(assertion: TestFormula<ChildFormula.Input, ChildFormula.Button>.() -> Unit) = apply {
-        childFormula.implementation.assertion()
-    }
-
-    fun assertOutput(on: ParentFormula.Output.() -> Unit) = apply {
-        observer.output(on)
-    }
-
-    class ParentFormula(
```
Comment (Laimiux) [inline formula-test/src/test/java/com/instacart/formula/test/TestFormulaRobot.kt:27]: Replacing all of this with a simpler test formula

File: formula-test/src/test/java/com/instacart/formula/test/TestFormulaTest.kt [hunk 3]
```diff
@@ -93,21 +128,11 @@ class TestFormulaTest {
...
-            .start()
-            .withTestFormula {
-                input(key = "child-key") { onChangeName("my name") }
-            }
-            .assertOutput {
-                assertThat(name).isEqualTo("my name")
-            }
-    }
-
-    @Test fun `input passed to formula`() {
```
Comment (Laimiux) [inline formula-test/src/test/java/com/instacart/formula/test/TestFormulaTest.kt:106]: Not needed

**Review by jasonostrander — APPROVED** [review]
I like this!

**Pull request merged**